atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/asts/bucketingthings/DefaultSweepAssignedBucketStore.java

```diff
@@ -0,0 +1,97 @@
+ */
+
+package com.palantir.atlasdb.sweep.asts.bucketingthings;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.CheckAndSetRequest;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.api.Value;
+import com.palantir.atlasdb.schema.generated.TargetedSweepTableFactory;
+import com.palantir.atlasdb.sweep.asts.bucketingthings.ObjectPersister.LogSafety;
+import com.palantir.conjure.java.serialization.ObjectMappers;
+import com.palantir.logsafe.exceptions.SafeIllegalStateException;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+final class DefaultSweepAssignedBucketStore implements SweepBucketAssignerStateMachineTable {
+    @VisibleForTesting
+    static final TableReference TABLE_REF =
+            TargetedSweepTableFactory.of().getSweepAssignedBucketsTable(null).getTableRef();
+
+    private final KeyValueService keyValueService;
+    private final ObjectPersister<BucketStateAndIdentifier> bucketStateAndIdentifierPersister;
+
+    private DefaultSweepAssignedBucketStore(
+            KeyValueService keyValueService,
+            ObjectPersister<BucketStateAndIdentifier> bucketStateAndIdentifierPersister) {
+        this.keyValueService = keyValueService;
+        this.bucketStateAndIdentifierPersister = bucketStateAndIdentifierPersister;
+    }
+
+    static DefaultSweepAssignedBucketStore create(KeyValueService keyValueService) {
+        // We explicitly don't want to fail on unknown properties;
+        ObjectMapper smileMapper = ObjectMappers.newClientSmileMapper();
+        return new DefaultSweepAssignedBucketStore(
+                keyValueService, ObjectPersister.of(smileMapper, BucketStateAndIdentifier.class, LogSafety.SAFE));
+    }
+
+    @Override
+    public void setInitialStateForBucketAssigner(long bucketIdentifier, long startTimestamp) {
+        BucketStateAndIdentifier initialState =
+                BucketStateAndIdentifier.of(bucketIdentifier, BucketAssignerState.start(startTimestamp));
+        Cell cell = SweepAssignedBucketStoreKeyPersister.INSTANCE.sweepBucketAssignerStateMachineCell();
+        casCell(cell, Optional.empty(), bucketStateAndIdentifierPersister.trySerialize(initialState));
+    }
+
+    @Override
+    public void updateStateMachineForBucketAssigner(
+            BucketStateAndIdentifier original, BucketStateAndIdentifier updated) {
+        Cell cell = SweepAssignedBucketStoreKeyPersister.INSTANCE.sweepBucketAssignerStateMachineCell();
+        casCell(
+                cell,
+                Optional.of(bucketStateAndIdentifierPersister.trySerialize(original)),
+                bucketStateAndIdentifierPersister.trySerialize(updated));
```

Comment:
please double-check: I could not find evidence that the persister is guaranteed to produce the elements in a defined order, which leads to fun times with this kind of thing. We might want to add `SORT_PROPERTIES_ALPHABETICALLY` to be 100% sure

Reply:
Yep, I'll also fix this in the bucket progress store from #7204